src/srml/balances.rs

```diff
@@ -0,0 +1,129 @@
+//! Implements support for the srml_balances module.
+use crate::{
+    codec::compact,
+    error::Error,
+    srml::system::System,
+    Client,
+    XtBuilder,
+};
+use futures::future::{
+    self,
+    Future,
+};
+use parity_scale_codec::Codec;
+use runtime_primitives::traits::{
+    MaybeSerializeDebug,
+    Member,
+    SimpleArithmetic,
+    StaticLookup,
+};
+use runtime_support::Parameter;
+use substrate_primitives::Pair;
+
+/// The subset of the `srml_balances::Trait` that a client must implement.
+pub trait Balances: System {
+    /// The balance of an account.
+    type Balance: Parameter
+        + Member
+        + SimpleArithmetic
+        + Codec
+        + Default
+        + Copy
+        + MaybeSerializeDebug
+        + From<<Self as System>::BlockNumber>;
+}
+
+/// The Balances extension trait for the Client.
+pub trait BalancesStore {
+    /// Balances type.
+    type Balances: Balances;
+
+    /// The 'free' balance of a given account.
+    ///
+    /// This is the only balance that matters in terms of most operations on
+    /// tokens. It alone is used to determine the balance when in the contract
+    ///  execution environment. When this balance falls below the value of
+    ///  `ExistentialDeposit`, then the 'current account' is deleted:
+    ///  specifically `FreeBalance`. Further, the `OnFreeBalanceZero` callback
+    /// is invoked, giving a chance to external modules to clean up data
+    /// associated with the deleted account.
+    ///
+    /// `system::AccountNonce` is also deleted if `ReservedBalance` is also
+    /// zero. It also gets collapsed to zero if it ever becomes less than
+    /// `ExistentialDeposit`.
+    fn free_balance(
+        &self,
+        account_id: <Self::Balances as System>::AccountId,
+    ) -> Box<dyn Future<Item = <Self::Balances as Balances>::Balance, Error = Error> + Send>;
+}
+
+impl<T: Balances + 'static> BalancesStore for Client<T> {
+    type Balances = T;
+
+    fn free_balance(
+        &self,
+        account_id: <Self::Balances as System>::AccountId,
+    ) -> Box<dyn Future<Item = <Self::Balances as Balances>::Balance, Error = Error> + Send>
+    {
+        let free_balance_map = || {
+            Ok(self
+                .metadata()
+                .module("Balances")?
+                .storage("FreeBalance")?
+                .get_map::<
+                <Self::Balances as System>::AccountId,
+                <Self::Balances as Balances>::Balance>()?)
```

Comment:
This formatting done by rustfmt? Looks odd

Reply:
that is rustfmt... it's not perfect...